src/components/Modals/SetAclPermissionsModal.jsx

```diff
@@ -0,0 +1,213 @@
+        clearInputFields();
+      }, 3000);
+    }
+  };
+
+  return (
+    <div>
+      <Dialog open={showModal} onClose={clearInputFields}>
+        <FormSection
+          title={
+            dataset.modalType === 'container'
+              ? 'Permission for all Documents'
+              : 'Permission for Document'
+          }
+          state={state}
+          statusType="Status"
+          defaultMessage="No action yet..."
+        >
+          <form onSubmit={handleAclPermission} autoComplete="off">
+            {dataset.modalType === 'document' ? (
+              <Typography
+                variant="subtitle2"
+                mb={2}
+                sx={{
+                  width: '250px',
+                  whiteSpace: 'nowrap',
+                  overflow: 'hidden',
+                  textOverflow: 'ellipsis'
+                }}
+              >
+                <span style={{ color: 'grey' }}>File Name: </span>
+                {dataset.documentName}
+              </Typography>
+            ) : null}
+            <FormControl required fullWidth sx={{ marginBottom: '1rem' }}>
+              <InputLabel id="permissionType-label">Select One</InputLabel>
+              <Select
+                labelId="permissionType-label"
+                id="permissionType"
+                label="Select One"
+                value={permissionState.permissionType}
+                onChange={(e) =>
+                  setPermissionState({ ...permissionState, permissionType: e.target.value })
+                }
+                name="setAclPerms"
+              >
+                <MenuItem value="Give">Give Permission</MenuItem>
+                <MenuItem value="Revoke">Revoke Permission</MenuItem>
+              </Select>
+            </FormControl>
+            <br />
+            <FormControl fullWidth sx={{ marginBottom: '1rem' }}>
+              <TextField
+                id="set-acl-to"
+                name="setAclTo"
+                value={permissionState.podUrlToSetPermissionsTo}
+                onChange={(e) =>
+                  setPermissionState({
+                    ...permissionState,
+                    podUrlToSetPermissionsTo: e.target.value
+                  })
+                }
+                placeholder={permissionState.podUrlToSetPermissionsTo}
+                label="Enter podURL"
+                required
+                error={permissionState.podUrlToSetPermissionsTo === podUrl}
+                helperText={
+                  permissionState.podUrlToSetPermissionsTo === podUrl
+                    ? 'Cannot modify your permissions to your own pod.'.toUpperCase()
+                    : ''
+                }
+              />
+            </FormControl>
+
+            {dataset.modalType === 'document' ? (
+              <DocumentSelection
+                htmlForAndIdProp="set-acl-doctype"
+                handleDocType={handleDocType}
+                docType={docType}
+              />
+            ) : null}
+
```

Comment:
I'm curious how much this `DocumentSelection` will be used here. Isn't the document type already set and this modal is to just change permissions?

Reply:
It looks like this component combines both permission modals together, so this would still be needed here.

But since that's the case, think we could use `&&` instead of a ternary with `?` and `null`.

Reply:
I BELIEVE what Andy is saying is that the modal that “needs” this, shouldn’t.  It’s the modal for setting permissions on a single file, correct?

That modal renders when the icon on the specific file is clicked, so we should be able to reference that specific file and pass it to the modal.  

This modal, the way it is now, should really only be necessary if we somehow got to it from an unspecified way - like just having a button at the top of the document list.

Reply:
Ah, I get it now. Given that the document is already selected per row, the document selection in this case is no longer necessary. I'm fine with that. We could actually remove this from this modal.

Reply:
Ok, I removed it